packages/common/http/src/headers.ts

```diff
@@ -258,6 +249,24 @@ export class HttpHeaders {
     }
   }
 
+  private allocateHeaderEntries(name: string, values: any) {
+    let headerValues: string[];
+
+    if (typeof values === 'string') {
+      headerValues = [values];
+    } else if (typeof values === 'number') {
+      headerValues = [values.toString()];
+    } else {
+      headerValues = values.map((value: any) => value.toString());
+    }
+
+    if (headerValues.length > 0) {
+      const key = name.toLowerCase();
+      this.headers.set(key, headerValues);
+      this.maybeSetNormalizedName(name, key);
+    }
+  }
```

Comment:
This can be simplified slightly, because;
1. The array will always have an item or more. 
2. No need to check for number/string.

```suggestion
const headerValues = (Array.isArray(values) ? values : [values]).map((value) => value.toString();
const key = name.toLowerCase();
this.headers.set(key, headerValues);
this.maybeSetNormalizedName(name, key);
```